agent/api/eniattachment.go

```diff
@@ -0,0 +1,94 @@
+// Copyright 2017 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License"). You may
+// not use this file except in compliance with the License. A copy of the
+// License is located at
+//
+//	http://aws.amazon.com/apache2.0/
+//
+// or in the "license" file accompanying this file. This file is distributed
+// on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+// express or implied. See the License for the specific language governing
+// permissions and limitations under the License.
+
+package api
+
+import (
+	"fmt"
+	"sync"
+	"time"
+
+	"github.com/aws/amazon-ecs-agent/agent/utils/ttime"
+	"github.com/cihub/seelog"
+	"github.com/pkg/errors"
+)
+
+// ENIAttachment contains the information of the eni attachment
+type ENIAttachment struct {
+	// TaskARN is the task identifier from ecs
+	TaskARN string `json:"taskArn"`
+	// AttachmentARN is the identifier for the eni attachment
+	AttachmentARN string `json:"attachmentArn"`
+	// AttachStatusSent indicates whether the attached status has been sent to backend
+	AttachStatusSent bool `json:"attachSent"`
+	// MACAddress is the mac address of eni
+	MACAddress string `json:"macAddress"`
+	// Status is the status of the eni: none/attached/detached
+	Status         ENIAttachmentStatus `json:"status"`
+	sentStatusLock sync.RWMutex
+	// ExpiresAt is the timestamp past which the ENI Attachment is considered
+	// unsuccessful. The SubmitTaskStateChange API, with the attachment information
+	// should be invoked before this timestamp.
+	ExpiresAt time.Time `json:"expiresAt"`
+	// ackTimer is used to register the expirtation timeout callback for unsuccessful
+	// ENI attachments
+	ackTimer ttime.Timer
```

Comment:
Not `ExpiresAt`. I was referring to `ackTimer`. I meant to say `unexported` by that. From https://golang.org/pkg/encoding/json/#Marshal: 

> Struct values encode as JSON objects. Each exported struct field becomes a member of the object, using the field name as the object key, unless the field is omitted for one of the reasons given below.